core/templates/services/rte-helper-modal.controller.spec.ts

```diff
@@ -158,4 +159,71 @@ describe('RteHelperModalComponent', () => {
       flush();
     }));
   });
+  describe('when there are validation errors in link form control', function() {
+    var customizationArgSpecs = [{
+      url: 'oppia.org',
+      text: 'oppia',
+    }];
+
+    beforeEach(() => {
+      fixture = TestBed.createComponent(RteHelperModalComponent);
+      component = fixture.componentInstance;
+      component.componentId = 'link',
+      component.attrsCustomizationArgsDict = {
+        heading: 'This value is not default.'
+      };
+      component.customizationArgSpecs = customizationArgSpecs;
+    });
+
+    it('should disable save button and display error message',
+      fakeAsync(() => {
+        component.ngOnInit();
+        flush();
+        component.customizationArgsForm.value[0] = 'oppia.org';
```

Comment:
How do we know what value[0] and value[1] are -- is there a way to do this that's based on the field names rather than relying on their ordering?

Reply:
Yes, that's right. Thanks for checking!